File: c#/GraphLinkedList.cs
using System.Collections.Generic;

namespace Graph
{
    public class GraphLinkedList
    {
        public enum GraphType
        {
            DIRECTED, // Graf terarah
            UNDIRECTED // Graf tidak terarah
        }
        public static int vertices;
        public static List<List<int>>? list;
        public static GraphType graphType;

        public GraphLinkedList(int vertices, GraphType graphType)
        {
            GraphLinkedList.vertices = vertices;
            GraphLinkedList.graphType = graphType;
            list = [];
            for (int i = 0; i < vertices; i++)
            {
                list.Add([]);
            }
        }

        public void AddEdge(int i, int j)
        {
            if (list != null)
            {
                list[i].Add(j);
                if (graphType == GraphType.UNDIRECTED)
                    list[j].Add(i);
            }
        }

        public void PrintGraph()
        {
            if (list != null)
                for (int i = 0; i < vertices; i++)
                {
                    Console.Write(i);
                    for (int j = 0; j < list[i].Count; j++)
                    {
                        Console.Write(" -> ");
                        Console.Write(list[i][j]);
                    }
                    Console.WriteLine();
                }
        }
    }
}

File: python/graphLinkedList.py
from typing import Literal

GraphType = Literal["directed", "undirected"]

class GraphLinkedList:
    vertices: int
    graphType: GraphType
    linkedList: list[list[int]]

    def __init__(self, vertices: int, graphType: GraphType) -> None:
        self.vertices = vertices
        self.graphType = graphType
        self.linkedList = [[]for _ in range(vertices)]
    
    def addEdge(self, i: int, j: int) -> None:
        self.linkedList[i].append(j)
        if self.graphType == "undirected":
            self.linkedList[j].append(i)
    
    def printGraph(self) -> None:
        for i in range(self.vertices):
            print(i, end='')
            for j in range(len(self.linkedList[i])):
                print(f" -> {self.linkedList[i][j]}", end='')
            print()
